src/crimson/os/seastore/seastore_types.h

```diff
@@ -1072,25 +1073,38 @@ struct rbm_alloc_delta_t {
     SET = 1,
     CLEAR = 2
   };
-  extent_types_t type;
   interval_set<blk_id_t> alloc_blk_ids;
   op_types_t op;
+  DENC(rbm_alloc_delta_t, v, p) {
+    DENC_START(1, 1, p);
+    denc(v.alloc_blk_ids, p);
+    denc(v.op, p);
+    DENC_FINISH(p);
+  }
 };
 
+paddr_t convert_blk_paddr_to_paddr(blk_paddr_t addr, size_t block_size,
+    uint32_t blocks_per_segment, device_id_t d_id);
+blk_paddr_t convert_paddr_to_blk_paddr(paddr_t addr, size_t block_size,
+	uint32_t blocks_per_segment);
```

Comment:
I see usages of blk_paddr_t in cache.cc and extent_placement_manager.cc.  I'd like to see the RandomBlockDevice interface converted to use paddr_t and all usages of blk_paddr_t restricted to the internals of RandomBlockDevice implementations, as with device_segment_id_t and SegmentManager implementations.

This may mean adjusting the paddr_t representation to remove the explicit segment component.